### PR TITLE
Add userId parameter to signOutUser

### PR DESCRIPTION
**Pull request by Aliandi**
Parity with C#

## Description
It was found that the TypeScript method signOutUser() differed from the C# implementation of [signOutUserAsync()](https://github.com/Microsoft/botbuilder-dotnet/blob/924eaa286fe3cebfaebae27570f504977c8591e9/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L642). The TypeScript version expects less parameters, not letting the user pass a userId.

_TypeScript_
`public async signOutUser(context: TurnContext, connectionName: string): Promise<void>`

_C#_
`public virtual async Task SignOutUserAsync(ITurnContext turnContext, string connectionName = null, string userId = null, CancellationToken cancellationToken = default(CancellationToken))`

## Specific Changes
  - add userId optional parameter to the method
  - add handling for when userId is null

### CODE REVIEW

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -393,13 +393,16 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
...
      * @param connectionName Name of the auth connection to use.
+     * @param userId id of user to sign out.
+     * @returns A promise that represents the work queued to execute.
      */
-    public async signOutUser(context: TurnContext, connectionName: string): Promise<void> {
+    public async signOutUser(context: TurnContext, connectionName?: string, userId?: string): Promise<void> {
         if (!context.activity.from || !context.activity.from.id) {
             throw new Error(`BotFrameworkAdapter.signOutUser(): missing from or from.id`);
         }
+        !userId? userId = context.activity.from.id: userId;
```
Comment (cleemullins) [inline libraries/botbuilder/src/botFrameworkAdapter.ts]: Could you do this using a less weird syntax? This makes my head hurt.

Comment (Aliandi) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: Done!

**Comment by munozemilio** [comment]
Hi @Aliandi,

Fork merges are disabled that's why the build is failing. Chris has grant you write permissions to the repo.
Could you please push your changes in a new branch from this repo?

Thanks!

**Comment by Aliandi** [comment]
Replaced by #837

**Pull request closed without merging**